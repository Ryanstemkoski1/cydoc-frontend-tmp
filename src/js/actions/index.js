import {ADD_NOTE, ADD_TEMPLATE, DATA_LOADED, LOGIN_REQUEST, LOGOUT, SAVE_NOTE} from "../constants/action-types";
import axios from 'axios'
import api from "../constants/api";

export function addNote(payload) {
    return { type: ADD_NOTE, payload };
}

export function addTemplate(payload) {
    return { type: ADD_TEMPLATE, payload: payload}
}

export function getAllRecords() {
    return function(dispatch) {
        return axios.get(api.records.dev)
            .then(response => response.data)
            .then(data => {
                dispatch({ type: DATA_LOADED, payload: data });
            });
    };
}

export function loginRequest(payload){
    return function(dispatch) {
        return axios.post(api.login.dev, payload)
            .then(res => {
                const user = res;
                console.log(user)
                localStorage.setItem('user', JSON.stringify(user));
                return user;
            })
            .then((user)=> {
                dispatch({type: LOGIN_REQUEST, payload: user.data})
                return user;
            })
            .catch(err => {
                return err.response;
            })
    }
}

export function logout(){
        localStorage.removeItem('user');
        return { type: LOGOUT };
}

export  function saveNote(payload){
    return function (dispatch) {
        const headers = {
            'Content-Type': 'application/json',

        };
        return axios.post(api.newRecord.dev,
            { headers: {
                'Content-Type': 'application/json'}},
            payload)
            .then(res => res.data)
            .then(data => {
                dispatch({type: SAVE_NOTE, payload: data})
            })
    }
}