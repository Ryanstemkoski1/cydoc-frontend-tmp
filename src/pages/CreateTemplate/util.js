import { questionTypes } from '../../constants/questionTypes';

// Filler text for when the root is NaN
export const NAN_QUESTION_TEXT =
    '(This question has no text, only follow-up questions)';

/**
 * Returns starting answerInfo object associated with the given type
 * @param {String} type: the question's response type
 */
export const getAnswerInfo = (type) => {
    let answerInfo;
    switch (type) {
        case questionTypes.YES_NO:
        case questionTypes.NO_YES:
            answerInfo = {
                yesResponse: '',
                noResponse: '',
            };
            break;
        case questionTypes.SHORT_TEXT:
        case questionTypes.NUMBER:
        case questionTypes.TIME:
        case questionTypes.BODY_LOCATION:
        case questionTypes.LIST_TEXT:
            answerInfo = {
                startResponse: '',
                endResponse: '',
            };
            break;
        case questionTypes.CLICK_BOXES:
            answerInfo = {
                startResponse: '',
                endResponse: '',
                options: ['', '', '', ''],
            };
            break;
        default:
            // Advanced type questions
            answerInfo = {
                options: [],
            };
    }
    return answerInfo;
};

/**
 * Returns a randomly generated node qid based on the disease code
 * @param {String} diseaseCode: 3-character abbreviation of the disease
 * @param {*} numQuestions: number of questions in the context
 */
export const createNodeId = (diseaseCode, numQuestions) => {
    const randomId = Math.floor(Math.random() * 9000000000) + 1000000000;
    const numZeros = 4 - numQuestions.toString().length;
    return (
        diseaseCode +
        '-' +
        randomId.toString() +
        '-' +
        '0'.repeat(numZeros) +
        numQuestions.toString()
    );
};

/**
 * Sorts the list of edges in place based on the edge's `toQuestionOrder`
 * @param {Array[Object]} edgeList
 * @param {Array[Object]} edges
 */
export const sortEdges = (edgeList, edges) => {
    edgeList.sort((a, b) => {
        const nodeA = parseFloat(edges[a].toQuestionOrder);
        const nodeB = parseFloat(edges[b].toQuestionOrder);

        if (nodeA < nodeB) {
            return -1;
        } else if (nodeA > nodeB) {
            return 1;
        } else {
            return 0;
        }
    });
};

/**
 * Given the node's responseType, text, and answerInfo, parse the text for the options,
 * if appropriate, and mutate the node's answerInfo directly.
 * Returns the parsed text.
 *
 * @param {String} responseType
 * @param {String} text
 * @param {Object} answerInfo
 * @param {String} category
 */
export const parseQuestionText = (responseType, text, answerInfo, category) => {
    if (text === 'nan') {
        text = `Root for ${category
            .replace('_', ' ')
            .toLowerCase()} ${NAN_QUESTION_TEXT}`;
    }

    if (
        responseType === questionTypes.CLICK_BOXES ||
        responseType.endsWith('POP') ||
        responseType === 'nan'
    ) {
        let click = text.search('CLICK');
        let selectStart = text.search('\\[');
        let selectEnd = text.search('\\]');
        let choices;
        if (click > -1) {
            // options are indicated by CLICK[...]
            choices = text.slice(click + 6, selectEnd);
            text = text.slice(0, click);
        } else {
            // options are indicated by [...]
            if (selectStart > 0) {
                choices = text.slice(selectStart + 1, selectEnd);
                text = text.slice(0, selectStart);
            }
        }
        choices = choices.split(',').map((response) => response.trim());
        answerInfo.options = choices;
    }
    return text;
};

/**
 * Replace all occurences of SYMPTOM or DISEASE with the given category
 *
 * @param {String} text
 * @param {String} category
 */
export const parsePlaceholder = (text, category) => {
    let placeholder = text.search(/SYMPTOM|DISEASE/);
    if (placeholder > -1) {
        text =
            text.substring(0, placeholder) +
            category.replace('_', ' ').toLowerCase() +
            text.substring(placeholder + 7);
    }
    return text;
};

/**
 * Adds blank{Yes, No, Template} to the appropriate fields in answerInfo
 */
export const setAnswerInfo = (answerInfo, node) => {
    if ('yesResponse' in answerInfo) {
        answerInfo.yesResponse = node.blankYes || '';
        answerInfo.noResponse = node.blankNo || '';
    } else if ('startResponse' in answerInfo) {
        const capturingRegex = /(?<startResponse>.*)\bANSWER\b((?<negStartResponse>.*)(\bNOTANSWER\b(?<negEndResponse>.*))|(?<endResponse>.*))/;
        const found = node.blankTemplate.match(capturingRegex);
        if (found) {
            answerInfo.startResponse = found.groups.startResponse || '';
            if (found.groups.endResponse) {
                answerInfo.endResponse = found.groups.endResponse || '';
            } else {
                answerInfo.endResponse = found.groups.negStartResponse || '';
                answerInfo.negEndResponse = found.groups.negEndResponse || '';
            }
        }
    }
    return answerInfo;
};

/**
 * Adds all direct children question of the given parent directly
 * to the graph object itself.
 *
 * Returns the updated numQuestions and nextEdgeID count.
 */
export const addChildrenNodes = (
    parentId,
    children,
    category,
    diseaseCode,
    graphData,
    contextData
) => {
    let {
        numQuestions,
        nextEdgeID,
        contextEdges,
        contextGraph,
        contextNodes,
    } = contextData;
    const { graph, edges, nodes } = graphData;

    // Create edges and nodes for every new question
    for (let edge of children) {
        const childId = createNodeId(diseaseCode, numQuestions);
        const nodeId = edges[edge].to;
        let responseType = nodes[nodeId].responseType;
        let text = nodes[nodeId].text;
        let answerInfo = getAnswerInfo(responseType);
        answerInfo = setAnswerInfo(answerInfo, nodes[nodeId]);

        // Preprocess the text to prepopulate the answerinfo if necessary
        // or replace instances of SYMPTOM and DISEASE
        text = parsePlaceholder(text, category);
        text = parseQuestionText(
            responseType,
            text,
            answerInfo,
            nodes[nodeId].category
        );
        contextNodes[childId] = {
            text,
            answerInfo,
            responseType,
            id: childId,
            order: numQuestions,
            hasChildren: graph[nodeId].length > 0,
            originalId: nodeId,
            hasChanged: false,
            parent: parentId,
        };

        contextEdges[nextEdgeID] = {
            from: parentId,
            to: childId,
        };

        contextGraph[childId] = [];
        contextGraph[parentId].push(nextEdgeID);
        nextEdgeID++;
        numQuestions++;
    }
    return { nextEdgeID, numQuestions };
};

/**
 * Walk up the knowledge graph and update all imported nodes
 * to hasChanged=true to indicate not to use the original ID anymore
 *
 * @param {Object} nodes
 * @param {String} qid
 */
export const updateParent = (nodes, qid) => {
    if (qid && !nodes[qid].hasChanged) {
        nodes[qid].hasChanged = true;
        updateParent(nodes, nodes[qid].parent);
    }
};

// Returns blank template for HPITemplateContext
export const getNewTemplate = () => {
    const rootId = createNodeId('000', 0);
    return {
        // `null` graphID indicates this is a newly created graph
        graphID: null,
        title: '',
        disease: '',
        bodySystem: '',
        numQuestions: 1,
        nextEdgeID: 0,
        hasChanged: true,
        root: rootId,
        graph: {
            [rootId]: [],
        },
        nodes: {
            [rootId]: {
                id: rootId,
                responseType: 'YES-NO',
                text: 'nan',
            },
        },
        edges: {},
        cydocGraphs: { graph: {}, edges: {}, nodes: {} },
    };
};
