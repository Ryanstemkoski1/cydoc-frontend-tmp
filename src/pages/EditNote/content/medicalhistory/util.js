/**
 * From https://stackoverflow.com/questions/38345273/javascript-algorithm-converting-to-roman-numbers
 */
const toRomanNumeral = (num) => {
    var arrConv = {
        0: { 1: 'I', 2: 'II', 3: 'III', 4: 'IV', 5: 'V', 6: 'VI', 7: 'VII', 8: 'VIII', 9: 'IX' },
        1: { 1: 'X', 2: 'XX', 3: 'XXX', 4: 'XL', 5: 'L', 6: 'LX', 7: 'LXX', 8: 'LXXX', 9: 'XC' },
        2: { 1: 'C', 2: 'CC', 3: 'CCC', 4: 'CD', 5: 'D', 6: 'DC', 7: 'DCC', 8: 'DCCC', 9: 'CM' },
        3: { 1: 'M', 2: 'MM', 3: 'MMM', 4: 'MMMM', 5: 'MMMMM', 6: 'MMMMMM', 7: 'MMMMMMM', 8: 'MMMMMMMM', 9: 'MMMMMMMMM' }
    },
    arr = num.toString().split("").reverse(),
    romansArray = arr.map(function (a, i) {
        return arrConv[i][a] || '';
    });

    return romansArray.reverse().join("");
}

/**
 * Replaces all numbers to roman numerals and all chars to lower case
 */
export const adjustValue = (val, mapping) => {
    let value = val.toLowerCase();
    if (value in mapping) {
        return mapping[value];
    }
    return val
        .replace(/\d+/gi, num => toRomanNumeral(num))
        .toLowerCase();
}
