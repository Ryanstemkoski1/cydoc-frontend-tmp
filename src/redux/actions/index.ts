import { ROSActionTypes } from './reviewOfSystemsActions';
import { MedicalHistoryActionTypes } from './medicalHistoryActions';
import { SurgicalHistoryActionTypes } from './surgicalHistoryActions';
import { MedicationsActionTypes } from './medicationsActions';
import { AllergiesActionTypes } from './allergiesActions';
import { SocialHistoryActionTypes } from './socialHistoryActions';
import { FamilyHistoryActionTypes } from './familyHistoryActions';
import { PhysicalExamActionTypes } from './physicalExamActions';
import { PlanActionTypes } from './planActions';
import { HpiActionTypes } from './hpiActions';
import { WidgetActionTypes } from './widgetActions';
import { chiefComplaintsActionTypes } from './chiefComplaintsActions';

export type AllActionTypes =
    | ROSActionTypes
    | PhysicalExamActionTypes
    | WidgetActionTypes
    | MedicalHistoryActionTypes
    | SurgicalHistoryActionTypes
    | MedicationsActionTypes
    | AllergiesActionTypes
    | SocialHistoryActionTypes
    | FamilyHistoryActionTypes
    | PlanActionTypes
    | HpiActionTypes
    | chiefComplaintsActionTypes;
