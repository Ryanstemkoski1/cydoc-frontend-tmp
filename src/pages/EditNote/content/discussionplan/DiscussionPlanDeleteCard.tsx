import React, { useState } from 'react';
import { connect } from 'react-redux';
import { deleteCondition } from 'redux/actions/planActions';
import { Button, Icon, Modal } from 'semantic-ui-react';

interface DispatchProps {
    deleteCondition: (conditionIndex: string) => void;
}

interface DeleteCardProps {
    uuid: string;
    name: string;
    index: number;
    deleteCurrent: (idx: number) => void;
}

export const FILLER_NAME = 'the currently viewed condition';

/**
 * Component for confirmation and deletion of the currently displayed condition
 * of the discussion and plan
 */
const DeleteCard = (props: DeleteCardProps & DispatchProps) => {
    const { uuid, name, index, deleteCurrent, deleteCondition } = props;
    const [isOpen, setIsOpen] = useState<boolean>(false);

    const confirmDelete = () => {
        deleteCondition(uuid);
        setIsOpen(false);
        deleteCurrent(index);
    };
    const onOpen = () => setIsOpen(true);
    const onClose = () => setIsOpen(false);

    return (
        <Modal
            size='mini'
            open={isOpen}
            onClose={onClose}
            className='plan-delete-modal'
            trigger={
                <Icon
                    onClick={onOpen}
                    name='trash alternate outline'
                    className='delete-btn'
                />
            }
        >
            <Modal.Content>
                {`Are you sure you want to delete the plan for ${
                    name || FILLER_NAME
                }?`}
            </Modal.Content>
            <Modal.Content className='btns'>
                <div className='modal-action'>
                    <Button
                        basic
                        color='grey'
                        content='Cancel'
                        onClick={onClose}
                    />
                    <Button
                        content='Delete'
                        onClick={confirmDelete}
                        floated='right'
                    />
                </div>
            </Modal.Content>
        </Modal>
    );
};

export default connect(null, { deleteCondition })(DeleteCard);
