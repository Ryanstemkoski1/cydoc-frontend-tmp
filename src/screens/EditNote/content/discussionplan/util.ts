/**
 * @fileoverview Miscellaneous helper functions and types/interfaces for the
 * discussion and plan page.
 */

import { PlanCondition } from '@redux/reducers/planReducer';
import { PlanActionTypes } from '@redux/actions/planActions';
import { WhenResponse, YesNoUncertainResponse } from '@constants/enums';

export interface DropdownOptions {
    value: string;
    label: string;
}

export type EventHandler = (_e: any, data: any) => void;

export type HandleOnAddItem = (_e: any, data: { [key: string]: any }) => void;

export type PlanAction = (
    conditionIndex: string,
    ...args: any[]
) => PlanActionTypes;

export type ConditionCategoryKey = keyof Omit<PlanCondition, 'name'>;

// Initial values with deterministic ids to allow for snapshot testing
export const conditionId = 'condition';
export const categoryId = 'category';
export const initialPlan = {
    conditions: {
        [conditionId]: {
            name: '',
            differentialDiagnoses: {
                [categoryId]: {
                    comments: '',
                    diagnosis: '',
                },
            },
            prescriptions: {
                [categoryId]: {
                    comments: '',
                    dose: '',
                    type: '',
                    signature: '',
                },
            },
            proceduresAndServices: {
                [categoryId]: {
                    comments: '',
                    procedure: '',
                    when: WhenResponse.None,
                },
            },
            referrals: {
                [categoryId]: {
                    comments: '',
                    department: '',
                    when: WhenResponse.None,
                },
            },
        },
    },
    survey: {
        admitToHospital: YesNoUncertainResponse.None,
        emergency: YesNoUncertainResponse.None,
        sicknessLevel: 0,
    },
};
