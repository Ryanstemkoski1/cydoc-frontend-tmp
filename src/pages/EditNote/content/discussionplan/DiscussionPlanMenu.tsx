import React, { useRef, useEffect, MouseEvent, ChangeEvent } from 'react';
import {
    Menu,
    Icon,
    Dropdown,
    Input,
    InputOnChangeData,
    MenuItemProps,
} from 'semantic-ui-react';
import DeleteCard from './DiscussionPlanDeleteCard';
import { connect } from 'react-redux';
import {
    PlanConditionsFlat,
    selectPlanConditions,
} from 'redux/selectors/planSelectors';
import { addCondition, updateConditionName } from 'redux/actions/planActions';
import { CurrentNoteState } from 'redux/reducers';
import { DISCUSSION_PLAN_MENU_BP } from 'constants/breakpoints.js';

const TAB_WIDTH = 250;

interface StateProps {
    conditions: PlanConditionsFlat[];
}

interface DispatchProps {
    addCondition: () => void;
    updateConditionName: (conditionIndex: string, newName: string) => void;
}

interface OwnProps {
    index: number;
    windowWidth: number;
    setCurrentIndex: (val: number) => void;
    setCurrentId: (val: string) => void;
}

type DiscussionPlanMenuProps = StateProps & DispatchProps & OwnProps;

/**
 * Hook for keeping track of previous values before rerender
 */
const usePrevious = <T,>(value: T) => {
    const ref = useRef<T>();
    useEffect(() => {
        ref.current = value;
    });
    return ref.current;
};

/**
 * Component for navigating between conditions for discussion and plan
 */
export const DiscussionPlanMenu = (props: DiscussionPlanMenuProps) => {
    const {
        windowWidth,
        index,
        conditions,
        setCurrentIndex,
        setCurrentId,
        addCondition,
        updateConditionName,
    } = props;

    const prevNumConditions = usePrevious<number>(conditions.length) || 0;

    const collapsed =
        windowWidth < DISCUSSION_PLAN_MENU_BP ||
        conditions.length * TAB_WIDTH > DISCUSSION_PLAN_MENU_BP;

    // Switch to the new tab if one was just added
    useEffect(() => {
        if (prevNumConditions < conditions.length) {
            const newest = conditions.length - 1;
            setCurrentId(conditions[newest].id);
            setCurrentIndex(newest);
        }
    }, [prevNumConditions, conditions.length]);

    // Select first tab on initial load
    useEffect(() => {
        if (conditions.length > 0) {
            setCurrentIndex(0);
            setCurrentId(conditions[0].id);
        }
    }, []);

    // Delete currently viewed tab and switch to the tab on the left, if there
    // exists one, otherwise switch to the right. If there are no more conditions,
    // empty the id.
    const deleteCurrent = (index: number) => {
        if (index === 0) {
            setCurrentId(conditions.length - 1 > 0 ? conditions[1].id : '');
            return;
        }
        setCurrentIndex(index - 1);
        setCurrentId(conditions[index - 1].id);
    };

    const selectTab = (
        _e: MouseEvent<HTMLAnchorElement>,
        { idx, uuid }: MenuItemProps
    ) => {
        setCurrentIndex(idx);
        setCurrentId(uuid);
    };

    const setName = (
        _e: ChangeEvent<HTMLInputElement>,
        { uuid, value }: InputOnChangeData
    ) => {
        updateConditionName(uuid, value);
    };

    const addButton = (
        <Menu.Item key='add-btn' id='add-condition' onClick={addCondition}>
            <Icon name='plus' />
        </Menu.Item>
    );

    // Show delete button only when there exist conditions to delete
    let deleteComponent;
    if (conditions.length > 0) {
        deleteComponent = (
            <Menu.Menu className='delete-btn-wrapper' position='right'>
                <DeleteCard
                    uuid={conditions[index].id}
                    name={conditions[index].name}
                    index={index}
                    deleteCurrent={deleteCurrent}
                />
            </Menu.Menu>
        );
    }

    // When screen is too small, show just the active condition as a tab
    // and hide the rest in a dropdown
    if (collapsed) {
        const dropdownItems = conditions.map((condition, i) => (
            <Menu.Item
                key={i}
                name={condition.name}
                text={condition.name}
                value={condition.name}
                active={index === i}
                idx={i}
                uuid={condition.id}
                onClick={selectTab}
            />
        ));
        dropdownItems.push(addButton);
        let currentTab;
        if (index < conditions.length) {
            currentTab = (
                <Menu.Item active>
                    <Input
                        transparent
                        type='text'
                        placeholder='Condition Name'
                        uuid={conditions[index].id}
                        value={conditions[index].name}
                        onChange={setName}
                    />
                </Menu.Item>
            );
        }

        return (
            <Menu tabular>
                {currentTab}
                <Menu.Item>
                    <Dropdown
                        value=''
                        icon='ellipsis horizontal'
                        options={dropdownItems}
                    />
                </Menu.Item>
                {deleteComponent}
            </Menu>
        );
    }

    // Otherwise render each condition as its own tab
    const menuItems = conditions.map((condition, i) => (
        <Menu.Item
            key={i}
            active={i === index}
            idx={i}
            uuid={condition.id}
            onClick={selectTab}
        >
            <Input
                transparent
                type='text'
                placeholder='Condition Name'
                value={condition.name}
                uuid={condition.id}
                idx={i}
                onChange={setName}
            />
        </Menu.Item>
    ));
    menuItems.push(addButton);
    return (
        <Menu stackable tabular>
            {menuItems}
            {deleteComponent}
        </Menu>
    );
};

export default connect(
    (state: CurrentNoteState) => ({ conditions: selectPlanConditions(state) }),
    { addCondition, updateConditionName }
)(DiscussionPlanMenu);
