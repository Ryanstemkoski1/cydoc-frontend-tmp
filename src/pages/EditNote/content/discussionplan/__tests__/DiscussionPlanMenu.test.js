import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import DiscussionPlanMenu from '../DiscussionPlanMenu';
import DiscussionPlanDeleteCard from '../DiscussionPlanDeleteCard';
import configureStore from 'redux-mock-store';
import { initialPlan, conditionId } from '../util';
import { Provider } from 'react-redux';
import { PLAN_ACTION as TYPES } from 'redux/actions/actionTypes';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore([]);

const mountWithStore = (discussionPlan = initialPlan, { ...props } = {}) => {
    const store = mockStore({ discussionPlan });

    props = {
        index: 0,
        setCurrentId: jest.fn(),
        setCurrentIndex: jest.fn(),
        ...props,
    };
    return {
        store,
        wrapper: mount(
            <Provider store={store}>
                <DiscussionPlanMenu {...props} />
            </Provider>
        ),
    };
};

describe('DiscussionPlanMenu', () => {
    it('renders without crashing', () => {
        const { wrapper } = mountWithStore();
        expect(wrapper).toBeTruthy();
    });

    it('matches snapshot', () => {
        const { wrapper } = mountWithStore();
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render delete button if condition length is greater than 0', () => {
        const { wrapper } = mountWithStore();
        expect(wrapper.find(DiscussionPlanDeleteCard)).toHaveLength(1);
    });

    it('should not render delete button if condition length is 0', () => {
        const { wrapper } = mountWithStore({
            conditions: {},
            survey: initialPlan.survey,
        });
        expect(wrapper.find(DiscussionPlanDeleteCard)).toHaveLength(0);
    });

    it('dispatches correct action and functions when new condition is added', () => {
        const mockSetCurrentId = jest.fn();
        const mockSetCurrentIndex = jest.fn();
        const { store, wrapper } = mountWithStore(undefined, {
            setCurrentId: mockSetCurrentId,
            setCurrentIndex: mockSetCurrentIndex,
        });
        wrapper.find('#add-condition').last().simulate('click');
        expect(mockSetCurrentId).toHaveBeenCalled();
        expect(mockSetCurrentIndex).toHaveBeenCalled();

        const expectedActions = [{ type: TYPES.ADD_CONDITION }];
        expect(store.getActions()).toEqual(expectedActions);
    });

    it('calls setCurrentIndex and setCurrentId when different tab selected', () => {
        const mockSetCurrentId = jest.fn();
        const mockSetCurrentIndex = jest.fn();
        const { wrapper } = mountWithStore(
            {
                ...initialPlan,
                conditions: {
                    ...initialPlan.conditions,
                    1: { ...initialPlan.conditions[conditionId] },
                },
            },
            {
                setCurrentId: mockSetCurrentId,
                setCurrentIndex: mockSetCurrentIndex,
            }
        );
        wrapper.find('a[uuid="1"]').simulate('click');
        expect(mockSetCurrentId).toHaveBeenCalled();
        expect(mockSetCurrentIndex).toHaveBeenCalled();
    });

    it('dispatches action with correct values on input change', () => {
        const { store, wrapper } = mountWithStore();
        const value = 'foo';
        wrapper.find('input[type="text"]').simulate('change', {
            target: { value },
        });

        const expectedActions = [
            {
                type: TYPES.UPDATE_CONDITION_NAME,
                payload: { newName: value, conditionIndex: conditionId },
            },
        ];
        expect(store.getActions()).toEqual(expectedActions);
    });
});
