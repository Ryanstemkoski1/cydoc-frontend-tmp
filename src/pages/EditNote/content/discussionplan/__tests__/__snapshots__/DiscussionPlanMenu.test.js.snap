// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscussionPlanMenu matches snapshot 1`] = `"<div class=\\"ui stackable tabular navMenu__single menu\\"><a idx=\\"0\\" uuid=\\"condition\\" class=\\"active item\\"><div uuid=\\"condition\\" idx=\\"0\\" class=\\"ui transparent input\\"><input placeholder=\\"Condition Name\\" type=\\"text\\" value=\\"\\"></div></a><a id=\\"add-condition\\" class=\\"item add-condition\\"><i aria-hidden=\\"true\\" class=\\"plus icon\\"></i></a><div class=\\"right menu delete-btn-wrapper\\"><i aria-hidden=\\"true\\" class=\\"trash alternate outline icon delete-btn\\"></i></div></div>"`;
